ai/providers/vertexai.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import os
+
+import google.api_core.exceptions
+import vertexai.generative_models
+
+from .base_provider import BaseAPIProvider
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+
+
+class VertexAPI(BaseAPIProvider):
+    VERTEX_AI_PROVIDER = "VertexAI"
+    MODELS = {
+        "gemini-1.5-flash-001": {
+            "name": "Gemini 1.5 Flash 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.5-flash-002": {
+            "name": "Gemini 1.5 Flash 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.5-pro-002": {
+            "name": "Gemini 1.5 Pro 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.5-pro-001": {
+            "name": "Gemini 1.5 Pro 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.0-pro-002": {
+            "name": "Gemini 1.0 Pro 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.0-pro-001": {
+            "name": "Gemini 1.0 Pro 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": False,
```

Comment:
This is a super nice approach and I'm glad it can be known within the models of this class 🙏